Sources/Views/MessageLabel.swift

```diff
@@ -313,10 +331,24 @@ open class MessageLabel: UILabel {
 
     private func parse(text: NSAttributedString) -> [NSTextCheckingResult] {
         guard enabledDetectors.isEmpty == false else { return [] }
-        let checkingTypes = enabledDetectors.reduce(0) { $0 | $1.textCheckingType.rawValue }
-        let detector = try? NSDataDetector(types: checkingTypes)
         let range = NSRange(location: 0, length: text.length)
-        let matches = detector?.matches(in: text.string, options: [], range: range) ?? []
+        var matches =  [NSTextCheckingResult]()
+
+        // Get matches of all .custom DetectorType and add it to matches array
+        let regexs = enabledDetectors
+            .filter { $0.isCustom }
+            .map { parseForMatches(with: $0, in: text, for: range) }
+            .joined()
+        matches.append(contentsOf: regexs)
```

Comment:
So it looks like we still parse all the regex and all the other detector types separately? I wonder if it's possible to register our own custom text checking type using `NSDataDetector`.

Fwiw, I love what's happening here -- I just have no time to benchmark what effect this will have on the framework and scroll performance. I have a serious suspicion --- but without benchmarks it's just suspicion